### PR TITLE
Feature/authentication and authorization

### PR DESCRIPTION
**Pull request by M-HAMMAD-HAZELSOFT**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,16 +1,49 @@
...
 // Register the AutoMapper
 builder.Services.AddAutoMapper(typeof(Program));
 
-// Register the UserService implementation
-// For IUserService interface as scoped service.
-builder.Services.AddScoped<IUserService, UserService>();
+// Register the ContactService implementation
+// For IContactService interface as scoped service.
+builder.Services.AddScoped<IContactService, ContactService>();
+
```
Comment (HassanMustafaHazel) [inline Program.cs:35]: again dependency injection.

File: Services/Authorization/AuthService.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            // creating the claims list for the JWT token
+            List<Claim> claims = new List<Claim>
+            {
+                new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
+                new Claim(ClaimTypes.Name, user.UserName),
+            };
+
+            // JWT private key from the config
+            SymmetricSecurityKey key = new SymmetricSecurityKey(
+                Encoding.UTF8.GetBytes(_configuration.GetSection("AppSettings:Token").Value)
```
Comment (HassanMustafaHazel) [inline Services/Authorization/AuthService.cs]: why magic strings are present.

**Review by HassanMustafaHazel — CHANGES_REQUESTED** [review]

File: Services/Authorization/AuthService.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                {
+                    Email = user.Email,
+                    UserName = user.UserName
+                };
+
+                // Creates a new User
+                var result = await _userManager.CreateAsync(newUser, user.Password);
+                if (result.Succeeded)
+                {
+                    await _signInManager.SignInAsync(newUser, isPersistent: false);
```
Comment (HassanMustafaHazel) [inline Services/Authorization/AuthService.cs]: why are you trying to login the user after creating it?

**Review by HassanMustafaHazel — CHANGES_REQUESTED** [review]

File: Services/Authorization/AuthService.cs (hunk 3 above)
Comment (M-HAMMAD-HAZELSOFT) [reply Services/Authorization/AuthService.cs]: removed it. I added it by mistake.

**Pull request merged**